src/method.rs

```diff
@@ -134,6 +138,30 @@ impl Method {
         }
     }
 
+    /// Convert static bytes into a `Method`.
+    pub const fn from_static(src: &'static [u8]) -> Method {
+        match src {
+            b"OPTIONS" => Method::OPTIONS,
+            b"GET" => Method::GET,
+            b"POST" => Method::POST,
+            b"PUT" => Method::PUT,
+            b"DELETE" => Method::DELETE,
+            b"HEAD" => Method::HEAD,
+            b"TRACE" => Method::TRACE,
+            b"CONNECT" => Method::CONNECT,
+            b"PATCH" => Method::PATCH,
+            src => {
+                if src.len() <= 15 {
+                    let inline = InlineExtension::from_static(src);
```

Comment:
With an `ExtensionStatic` available, is there a benefit to keeping this one?

Reply:
`InlineExtension` has no heap allocation and has the data available in the same allocation. Most HTTP methods fit in this case, and it has no overhead.

`ExtensionStatic` is a niche for unusually long methods, but has a heap allocation and an extra indirection.

We have three options here:

- Keep both. No real downsides aside from another internal type, and does not increase the size of the `Method` type.
- Keep only `InlineExtension`, restricting `from_static` to only methods of `len <= 15`.
- Keep only `ExtensionStatic`, forcing any `from_static` method to use a heap allocation and an extra indirection.

Personally, I prefer the first choice the most, and the third one the least (since it introduces pointless overhead for most WebDAV methods, albeit minor).